pkg/dependency/parser/golang/mod/parse.go

```diff
@@ -80,7 +82,21 @@ func (p *Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependenc
 
 	skipIndirect := true
 	if modFileParsed.Go != nil { // Old go.mod file may not include the go version. Go version for these files  is less than 1.17
-		skipIndirect = lessThan117(modFileParsed.Go.Version)
+		skipIndirect = lessThan(modFileParsed.Go.Version, 1, 17)
+	}
+
+	// Use minimal required go version from `toolchain` line (or from `go` line if `toolchain` is omitted) as `stdlib`.
+	// Show `stdlib` only with `useMinVersion` flag.
+	if toolchainVer := toolchainVersion(modFileParsed.Toolchain, modFileParsed.Go); p.useMinVersion && toolchainVer != "" {
```

Comment:
nit: It's no big deal, but if `useMinVersion == false`, we don't need to get the toolchain version. We may want to evaluate `useMinVersion` first before getting the toolchain version.

Reply:
It make sense.
Updated in https://github.com/aquasecurity/trivy/pull/7163/commits/e6411c054e1c8bcf2fdf4175c0a7b522698d3f91